common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.identity.common;
+
+import android.os.Environment;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class CodeMarkerManager {
+
+    private static boolean enableCodeMarker = false;
+    private static int MAX_SIZE_CODE_MARKER = 1000;
+    private static volatile List<CodeMarker> codeMarkers = new ArrayList<CodeMarker>();
+    private static long baseMilliSeconds = 0;
+    public static void markCode(int marker) {
+        if(enableCodeMarker) {
+            if(codeMarkers.size() >= MAX_SIZE_CODE_MARKER) {
+                clear();
+            }
+            long currentMilliSeconds = System.currentTimeMillis();
+            if (codeMarkers.size() == 0) {
+                baseMilliSeconds = currentMilliSeconds;
+            }
+            SimpleDateFormat f = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
+            codeMarkers.add(new CodeMarker(marker, currentMilliSeconds - baseMilliSeconds, f.format(new Date()), Thread.currentThread().getId()));
```

Comment:
When would the markers be popped out of this list? I see it happens if we hit max size - is that the only time?

Reply:
As per the new iteration I have pushed, the markers will also be cleared on completion of the test cases "TestCasePerf" and "TestCasePerfBrokerHost".

Reply:
Hi @shahzaibj ,
Please look at the resolution of this comment. Thanks.